packages/flutter_test/lib/src/finders.dart

```diff
@@ -460,57 +498,406 @@ class CommonFinders {
   }
 }
 
-/// Searches a widget tree and returns nodes that match a particular
-/// pattern.
-abstract class Finder {
-  /// Initializes a Finder. Used by subclasses to initialize the [skipOffstage]
-  /// property.
-  Finder({ this.skipOffstage = true });
 
-  /// Describes what the finder is looking for. The description should be
-  /// a brief English noun phrase describing the finder's pattern.
-  String get description;
+/// Provides lightweight syntax for getting frequently used semantics finders.
+///
+/// This class is instantiated once, as [CommonFinders.semantics], under [find].
+class CommonSemanticsFinders {
+  const CommonSemanticsFinders._();
 
-  /// Returns all the elements in the given list that match this
-  /// finder's pattern.
+  /// Finds an ancestor of [of] that matches [matching].
   ///
-  /// When implementing your own Finders that inherit directly from
-  /// [Finder], this is the main method to override. If your finder
+  /// If [matchRoot] is true, then the results of [of] are included in the
+  /// search and results.
+  FinderBase<SemanticsNode> ancestor({
```

Comment:
It's more a natural consequence of what the Ancestor and Descendent finders _are_. The `SemanticsFinder` and `WidgetFinder` classes primarily define the `allCandidates` getter. The Ancestor and Descendant finders override and change that behavior to be completely dependent on the finders that are passed into it.

Forcing it to inherit from `SemanticsFinder`/`WidgetFinder` requires weird workarounds for some of the additional things those classes define, like the `WidgetFinder`'s `skipOffstage` or the `SemanticsFinder`'s `root`. They don't actually make sense in the context of an ancestor/descendant finder, but they'd need to be implemented to support the typing.

The Widget versions of these classes do inherit from `WidgetFinder` because of backwards compatibility requirements, but they're awkward at best.